testsuite/tests/typing-jkind-bounds/subsumption/quality.ml

```diff
@@ -0,0 +1,251 @@
+(* TEST
+    flags = "-extension layouts_alpha";
+    expect;
+*)
+
+(* Test that not-best kinds are respected *)
+
+module _ : sig
+  type 'a t : immutable_data with 'a
+end = struct
+  type 'a t
+end
+[%%expect {|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type 'a t
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type 'a t end
+       is not included in
+         sig type 'a t : immutable_data with 'a end
+       Type declarations do not match:
+         type 'a t
+       is not included in
+         type 'a t : immutable_data with 'a
+       The kind of the first is value
+         because of the definition of t at line 4, characters 2-11.
+       But the kind of the first must be a subkind of immutable_data with 'a
+         because of the definition of t at line 2, characters 2-36.
+|}]
+
+(* This appears sound to accept. But it isn't. See the following test. *)
```

Comment:
This doesn't appear sound to be. In the struct, we have `t : value`, with no mode-crossing ability whatsoever. But not so in the sig. Maybe there is a different relationship you meant to test?

Reply:
This is indeed testing what I wanted to - I think maybe this is just more obvious to you, and I should update the comment. I was temporarily under the impression that abstract types could be given `best` if the signature is immediately implemented. But this is clearly wrong.

Reply:
I'll admit being completely confused about what this file is testing, because there were lots of subsumption checks between e.g. `value <: value mod global with a` -- which to my eyes clearly shouldn't work. But of course it only doesn't work because we (rightly) refuse to expand `a`. This only became apparent to me later when the type in the with-bounds is more complex, so I had to think harder.

Bottom line: your response above is right. Abstract types "obviously" don't expand, and so I didn't see what was even being tested. But this was my mistake.